src/main/java/io/github/intoto/dsse/helpers/SimpleSigstoreSigner.java

```diff
@@ -0,0 +1,58 @@
+package io.github.intoto.dsse.helpers;
+
+import dev.sigstore.KeylessSigner;
+import dev.sigstore.KeylessSignerException;
+import dev.sigstore.bundle.Bundle;
+import io.github.intoto.dsse.models.Signer;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Optional;
+
+public class SimpleSigstoreSigner implements Signer {
+    private String keyId;
+    Optional<Bundle.DSSESignature> dsseSignature;
+    Bundle result;
+
+    public byte[] sign(byte[] payload) throws InvalidAlgorithmParameterException, CertificateException,  IOException, NoSuchAlgorithmException, InvalidKeySpecException, InvalidKeyException, KeylessSignerException {
+        KeylessSigner functionary = new KeylessSigner.Builder().sigstorePublicDefaults().build();
+        this.result = functionary.sign(payload);
+
+        // set keyId
+        X509Certificate certificate = (X509Certificate) (this.result.getCertPath().getCertificates().getFirst());
+        String oid = "1.3.6.1.4.1.57264.1.8";
+        byte[] extensionValue = certificate.getExtensionValue(oid);
+        String issuer = new String(extensionValue, StandardCharsets.UTF_8);
+        this.keyId = issuer.substring(4);
+        Object subAltArr = certificate.getSubjectAlternativeNames().toArray()[0];
+        String subAltName = subAltArr.toString();
+        subAltName = subAltName.substring(4, subAltName.length() - 1);
+        this.keyId = keyId.concat(" " + subAltName);
```

Comment:
can you compute the keyid such as:

`<oidc provider>:<SAN>` ? this way we can also know who actually provided the identity to fulcio

Reply:
Is this how it's supposed to look?
`<https://accounts.example.com>: <alice@example.com>`